lib/config/guiconfiguration.go

```diff
@@ -26,42 +28,27 @@ type GUIConfiguration struct {
 	InsecureAllowFrameLoading bool   `xml:"insecureAllowFrameLoading,omitempty" json:"insecureAllowFrameLoading"`
 }
 
-func (c GUIConfiguration) Address() string {
-	if override := os.Getenv("STGUIADDRESS"); override != "" {
-		// This value may be of the form "scheme://address:port" or just
-		// "address:port". We need to chop off the scheme. We try to parse it as
-		// an URL if it contains a slash. If that fails, return it as is and let
-		// some other error handling handle it.
-
-		if strings.Contains(override, "/") {
-			url, err := url.Parse(override)
-			if err != nil {
-				return override
-			}
-			return url.Host
+func GUIConfigFromString(envAddr string) (c GUIConfiguration) {
```

Comment:
I think you need to run `c` through `util.SetDefaults` first, otherwise it will e.g. be disabled due as all members will be initialized at their zero values.

Reply:
Good catch, the fact that the author hasn't caught this gets me a bit worried about the level of testing done.

Reply:
That seems a bit harsh, there is still lots of reviewing going on and tests don't even pass yet. I am positive in that stadium lots of my PRs have errors of that magnitude in them, exactly because there wasn't much (any) real word testing done yet.

Reply:
Agree, perhaps it's not yet tested as it's deemed early, so I am too quick to jump to conclusions.

Reply:
Indeed. I would have caught this bug if I had tested the `STGUIADDRESS(ES)` environment overriding feature. The point is that as this stadium, my validation procedure is still very light, given the yet high rate of review activity on this PR. I'll try to harden a bit my validation steps.